doc/source/contributing/syntax_guide.md

```diff
@@ -0,0 +1,201 @@
+(syntax_guide)=
+# Syntax Guide
+
+## Adding targets
```

Comment:
doubling down on the reasoning point. It is great too highlight and show here the syntax for the more common things like anchors or toctree, however, again much more emphasis on why it's important or even imprescindible and I think all sections should have also external references. This could for example link to https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#hyperlink-targets which extends and gives more detail on the syntax (like explaining that it can be used to link to list items or paragraphs in addition to sections) and/or https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#ref-role.

Reply:
I don't think this is resolved. It still says linking to headings only which is the most common thing but not the only one as shown in the first reference I linked. We could consider using a seealso directive at the end of the section to add all relevant links.